base/src/main/java/com/smartdevicelink/managers/screen/BaseTextAndGraphicManager.java

```diff
@@ -154,821 +142,353 @@ public void dispose(){
-			if (i > 0) {
-				showString1.append(" - ").append(showFields.get(i));
-			}else{
-				showString1.append(showFields.get(i));
-			}
-		}
-		show.setMainField1(showString1.toString());
-
-		MetadataTags tags = new MetadataTags();
-		tags.setMainField1(findNonNullMetadataFields());
-
-		show.setMetadataTags(tags);
-
-		return show;
-	}
-
-	private Show assembleTwoLineShowText(Show show){
-
-		StringBuilder tempString = new StringBuilder();
-		MetadataTags tags = new MetadataTags();
-
-		if (textField1 != null && textField1.length() > 0) {
-			tempString.append(textField1);
-			if (textField1Type != null){
-				tags.setMainField1(textField1Type);
-			}
-		}
-
-		if (textField2 != null && textField2.length() > 0) {
-			if (( textField3 == null || !(textField3.length() > 0)) && (textField4 == null || !(textField4.length() > 0))){
-				// text does not exist in slots 3 or 4, put text2 in slot 2
-				show.setMainField2(textField2);
-				if (textField2Type != null){
-					tags.setMainField2(textField2Type);
-				}
-			} else if (textField1 != null && textField1.length() > 0) {
-				// If text 1 exists, put it in slot 1 formatted
-				tempString.append(" - ").append(textField2);
-				if (textField2Type != null){
-					List<MetadataType> typeList = new ArrayList<>();
-					typeList.add(textField2Type);
-					if (textField1Type != null){
-						typeList.add(textField1Type);
-					}
-					tags.setMainField1(typeList);
-				}
-			}else {
-				// If text 1 does not exist, put it in slot 1 unformatted
-				tempString.append(textField2);
-				if (textField2Type != null){
-					tags.setMainField1(textField2Type);
-				}
-			}
-		}
-
-		// set mainfield 1
-		show.setMainField1(tempString.toString());
-
-		// new stringbuilder object
-		tempString = new StringBuilder();
-
-		if (textField3 != null && textField3.length() > 0){
-			// If text 3 exists, put it in slot 2
-			tempString.append(textField3);
-			if (textField3Type != null){
-				List<MetadataType> typeList = new ArrayList<>();
-				typeList.add(textField3Type);
-				tags.setMainField2(typeList);
-			}
-		}
-
-		if (textField4 != null && textField4.length() > 0){
-			if (textField3 != null && textField3.length() > 0){
-				// If text 3 exists, put it in slot 2 formatted
-				tempString.append(" - ").append(textField4);
-				if (textField4Type != null){
-					List<MetadataType> typeList = new ArrayList<>();
-					typeList.add(textField4Type);
-					if (textField3Type != null){
-						typeList.add(textField3Type);
-					}
-					tags.setMainField2(typeList);
-				}
-			} else {
-				// If text 3 does not exist, put it in slot 3 unformatted
-				tempString.append(textField4);
-				if (textField4Type != null){
-					tags.setMainField2(textField4Type);
+	private synchronized void sdlUpdate(final CompletionListener listener) {
+		if (transactionQueue.getTasksAsList().size() > 0) {
+			//Transactions already exist, cancelling them
+			for (Task task : transactionQueue.getTasksAsList()) {
```

Comment:
This is done so all Task will start. The task will cancel right away and call back after starting then complete the developer's listener back. I tested rapid-firing updates and found this to be the best way. I think this is the best way, to ensure the listener sent with it is only called back once with the correct value. Also, this is how IOS handles its operations. It would be ideal if we could change the state of the operation in TaskMaster to canceled and the operation still started and allow the operation itself to handle stopping itself.